browser/lib/transport/xhrrequest.js

```diff
@@ -115,7 +115,7 @@ var XHRRequest = (function() {
 
 		if(body) {
 			var contentType = headers['content-type'] || (headers['content-type'] = 'application/json');
-			if(contentType == 'application/json' && typeof(body) != 'string')
```

Comment:
Line 112 is the `accept` header, not `content-type`; it's specifying what content types it would be happy for the server to return, rather than the content type of the body it's sending. For auth requests ably-js sets the `accept` header to `application/json, text/plain` (since some auth servers return a json tokenDetails and some return just a token string), hence the use of indexOf there.